### ISSUE TITLE
Repo Gardening / Turnstile: avoid multiple action runs on issue creation / labeling

### ISSUE THREAD

**Issue by jeherve** [Automattic/jetpack#40088]
#### Steps to reproduce

1. Go to the Issues tab and start creating a new blank issue.
2. Fill in the issue title and contents, and add 4 labels to the issue: "Actions", "[Action] Repo Gardening", "[Pri] Normal", "[Type] Bug".
4. Hit the "Create" button to validate and create a new issue.

At that point, 5 different events are triggered:

- An event for the issue creation
- An event for each of the labels added to the issue.

Each one of those events triggers a run of the Gardening action:
https://github.com/Automattic/jetpack/blob/5e41cdf798b5872dfed14c7387aac2a727c6657e/.github/workflows/gardening.yml#L5-L6

![Image](https://github.com/user-attachments/assets/6f8d3260-17e8-4a22-a7cb-86a613a7c85f)

Since #33189, such events do not trigger [Turnstile](https://github.com/Automattic/jetpack/tree/5e41cdf798b5872dfed14c7387aac2a727c6657e/.github/actions/turnstile), and we consequently do not wait for each run to complete before to start the next one.

This is problematic when the Gardening action makes an action like post to Slack: multiple messages end up being posted on Slack.

@anomiex Would it be possible to group the runs by event type? All `issues` runs would wait on each other, all `push` events would wait on each other, but a `push` event wouldn't have to wait on issues being created for example.

**Comment by anomiex** [comment]
That's supposed to be what's already happening. If the event is "issues" or "issue_comment" it should be grouping those together, and if it's not one of those two (e.g. it's "push") it should be ignoring any of those.

Ah, I think I see the problem, or at least one problem.
```
Run ./.github/actions/turnstile
  with:
    events: issues issue_comment
    ignore-events: issues issue_comment
    poll-interval: 60
    token: ***
```
It seems it's interpreting the `''` in `ignore-events: ${{ ( github.event_name == 'issues' || github.event_name == 'issue_comment' ) && '' || 'issues issue_comment' }}` as falsey and so going on to do the `||`. I wish Actions had a proper ternary operator.
